linker is trying to read build/lib/ as a file, but it's actually a directory

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug
Here is a clear and concise description of what the problem is:
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->

Building from master on Arch Linux using the [AUR PKGBUILD](https://aur.archlinux.org/packages/kodi-git) fails without reverting https://github.com/xbmc/xbmc/commit/42aab8bf10adc59bac7715fac63e68cd4a6d6a99
```
[ 98%] Linking CXX static library libkodi.a
[ 98%] Built target libkodi
[ 98%] Building CXX object CMakeFiles/kodi.dir/xbmc/platform/posix/main.cpp.o
[100%] Linking CXX executable kodi.bin
/usr/bin/ld.gold: fatal error: build/lib/: pread failed: Is a directory
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/kodi.dir/build.make:588: kodi.bin] Error 1
make[1]: *** [CMakeFiles/Makefile2:7431: CMakeFiles/kodi.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

I discovered this with a `git bisect` @fuzzard

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
<!--- Tell us what should happen -->
<!--- Put your text below this line -->

Build should complete


## Actual Behavior
<!--- Tell us what happens instead -->
<!--- Put your text below this line -->

Build fails at linking

## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->

?

### To Reproduce
Steps to reproduce the behavior:
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->
1. Attempt to build with aforementioned PKGBUILD (although current version applies a revert patch which needs to be commented out to trigger the error
```
wget  https://aur.archlinux.org/cgit/aur.git/snapshot/kodi-git.tar.gz
tar zxf kodi-git.tar.gz
cd kodi-git
sed '/patch -p1/d' PKGBUILD
makepkg -s
```

### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
The debuglog can be found here:

N/A

### Screenshots 
Here are some links or screenshots to help explain the problem:
<!--- Put your text below this line -->

N/A

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- Put your text below this line -->



### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Arch Linux
 - Kodi version: latest from HEAD



<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

--- a/cmake/modules/FindFstrcmp.cmake
+++ b/cmake/modules/FindFstrcmp.cmake
@@ -12,11 +12,11 @@ if(NOT TARGET ${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME})
   include(cmake/scripts/common/ModuleHelpers.cmake)
 
   set(${CMAKE_FIND_PACKAGE_NAME}_MODULE_LC fstrcmp)
-  set(${${CMAKE_FIND_PACKAGE_NAME}_MODULE_LC}_DISABLE_VERSION ON)
 
   SETUP_BUILD_VARS()
 
   if(ENABLE_INTERNAL_FSTRCMP)
+    message(STATUS "Building ${${CMAKE_FIND_PACKAGE_NAME}_MODULE_LC}: \(version \"${${${CMAKE_FIND_PACKAGE_NAME}_MODULE}_VER}\"\)")
     find_program(LIBTOOL libtool REQUIRED)
 
     find_program(AUTORECONF autoreconf REQUIRED)

